src/operator/subgraph/mkldnn/mkldnn_conv.cc

```diff
@@ -0,0 +1,325 @@
+            inputs[in_gamma], inputs[in_beta], inputs[in_mean], inputs[in_var],
+            cached_weight_, cached_bias_, bn_param);
+      });
+    }
+    std::vector<NDArray> new_inputs;
+    std::vector<OpReqType> new_req;
+    std::vector<NDArray> new_outputs;
+    if (with_bn) {
+      new_inputs = {inputs[in_data], *cached_weight_, *cached_bias_};
+      new_req = {req[in_data], req[in_weight], req[in_beta]};
+    } else {
+      if (conv_params.no_bias) {
+        new_inputs = {inputs[in_data], inputs[in_weight]};
+        new_req = {req[in_data], req[in_weight]};
+      } else {
+        new_inputs = {inputs[in_data], inputs[in_weight], inputs[in_bias]};
+        new_req = {req[in_data], req[in_weight], req[in_bias]};
+      }
+    }
+    if (with_sum)
+      new_outputs = {inputs[in_sum]};
+    else
+      new_outputs = {output};
+    ConvolutionFusionComputeExCPU(*conv_attrs_, ctx, new_inputs, new_req,
+                                  new_outputs);
+  }
+
+  OpStatePtr CreateSgMKLDNNConvOpState(const nnvm::NodeAttrs &attrs, Context ctx,
+                                       const std::vector<TShape> &in_shapes,
+                                       const std::vector<int> &in_types) {
+    return OpStatePtr::Create<SgMKLDNNConvOperator>(attrs);
+  }
+
+  void SgMKLDNNConvOpForward(const OpStatePtr &state_ptr, const OpContext &ctx,
+                             const std::vector<NDArray> &inputs,
+                             const std::vector<OpReqType> &req,
+                             const std::vector<NDArray> &outputs) {
+    SgMKLDNNConvOperator &op = state_ptr.get_state<SgMKLDNNConvOperator>();
+    op.Forward(ctx, inputs, req, outputs);
+  }
+
+NNVM_REGISTER_OP(_sg_mkldnn_conv)
+.describe(R"code(_sg_mkldnn_conv)code" ADD_FILELINE)
+.set_num_inputs(DefaultSubgraphOpNumInputs)
+.set_num_outputs(DefaultSubgraphOpNumOutputs)
+.set_attr<nnvm::FListInputNames>("FListInputNames",
+                                  DefaultSubgraphOpListInputs)
+.set_attr<nnvm::FListOutputNames>("FListOutputNames",
+  [](const NodeAttrs& attrs) {
+  return std::vector<std::string>{"output"};
+})
+.set_attr<FCreateOpState>("FCreateOpState", CreateSgMKLDNNConvOpState)
+.set_attr<nnvm::FInferShape>("FInferShape", DefaultSubgraphOpShape)
+.set_attr<nnvm::FInferType>("FInferType", DefaultSubgraphOpType)
+.set_attr<FInferStorageType>("FInferStorageType",
+                              DefaultSubgraphOpStorageType)
+.set_attr<FStatefulComputeEx>("FStatefulComputeEx<cpu>",
+                              SgMKLDNNConvOpForward)
+.set_attr<nnvm::FMutateInputs>("FMutateInputs",
+                                DefaultSubgraphOpMutableInputs)
+.set_attr<FResourceRequest>("FResourceRequest",
+                            DefaultSubgraphOpResourceRequest)
+.set_attr<std::string>("key_var_num_args", "num_args")
+.set_attr<nnvm::FInplaceOption>("FInplaceOption", [](const nnvm::NodeAttrs
+                                                          &attrs) {
```

Comment:
If you specify the return type, like `-> std::vector<std::pair<int, int>>` then you can do things like return {}; further in the lambda or just `return std::make_pair(0, 0)`.